File: packages/attachment_engine/attachment_engine_android/android/src/main/kotlin/com/dhc/attachment_engine/PathsChannel.kt
package com.dhc.attachment_engine

import android.content.Context
import io.flutter.plugin.common.BinaryMessenger

/**
 * Replaces `path_provider`: exposes app-private storage directories.
 * Implements the Pigeon-generated [PathsHostApi].
 */
class PathsChannel(private val context: Context) : PathsHostApi {
    fun register(messenger: BinaryMessenger) {
        PathsHostApi.setUp(messenger, this)
    }

    fun unregister(messenger: BinaryMessenger) {
        PathsHostApi.setUp(messenger, null)
    }

    override suspend fun getApplicationSupportDirectory(): String = context.filesDir.absolutePath

    override suspend fun getApplicationCacheDirectory(): String = context.cacheDir.absolutePath
}

File: packages/attachment_engine/attachment_engine_ios/ios/attachment_engine_ios/Sources/attachment_engine_ios/PathsChannel.swift
import Flutter
import Foundation

/// Replaces `path_provider`: exposes app-private storage directories via
/// `NSSearchPathForDirectoriesInDomains`. Implements the Pigeon-generated
/// `PathsHostApi`.
class PathsChannel: NSObject, PathsHostApi {
  func register(with messenger: FlutterBinaryMessenger) {
    PathsHostApiSetup.setUp(binaryMessenger: messenger, api: self)
  }

  func unregister(with messenger: FlutterBinaryMessenger) {
    PathsHostApiSetup.setUp(binaryMessenger: messenger, api: nil)
  }

  func getApplicationSupportDirectory() async throws -> String {
    let paths = NSSearchPathForDirectoriesInDomains(
      .applicationSupportDirectory, .userDomainMask, true)
    let dir = paths.first ?? NSTemporaryDirectory()
    try? FileManager.default.createDirectory(
      atPath: dir, withIntermediateDirectories: true, attributes: nil)
    return dir
  }

  func getApplicationCacheDirectory() async throws -> String {
    let paths = NSSearchPathForDirectoriesInDomains(.cachesDirectory, .userDomainMask, true)
    let dir = paths.first ?? NSTemporaryDirectory()
    try? FileManager.default.createDirectory(
      atPath: dir, withIntermediateDirectories: true, attributes: nil)
    return dir
  }
}
